### PR TITLE
Switch from String to Text

### PR DESCRIPTION
**Pull request by tfausak**
This switches as many things as possible from `String` to `Text`. Some strings stuck around because file paths and time parsing/formatting require them. But rending CommonMark is by far the bulk of the work here and it requires text.

### CODE REVIEW

**Comment by tfausak** [comment]
This improves performance in a few ways:

- Total allocations in bytes fell from 131,305,548 to 52,055,992. So this only allocates 40% as much as the `String`-based version.
- Total memory in use fell from 28 MB to 4 MB. That's 14%. 
- Productivity raised from 89% to 99%. 
- Total time in seconds fell from 0.47 to 0.42. That's 90% of the run time.
- `renderPieces`, `splitOn`, and `commonMark` all disappeared from the cost center table. The only thing that spends any time or allocates any memory is `writeFileAt`. 

TL;DR: Uses way less memory and is faster. Also arguably more correct.
